onnxruntime/core/providers/webnn/builders/helper.h

```diff
@@ -128,90 +133,106 @@ std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_v
+    {"GlobalMaxPool", {"maxPool2d", true}},
+    {"GlobalLpPool", {"l2Pool2d", false}},
+    {"Greater", {"greater", false}},
+    {"GreaterOrEqual", {"greaterOrEqual", false}},
+    {"GroupNormalization", {"meanVarianceNormalization", false}},
+    {"HardSigmoid", {"hardSigmoid", false}},
+    {"HardSwish", {"hardSwish", true}},
+    {"Identity", {"identity", false}},
+    {"InstanceNormalization", {"meanVarianceNormalization", false}},
+    {"LayerNormalization", {"meanVarianceNormalization", false}},
+    {"LeakyRelu", {"leakyRelu", true}},
+    {"Less", {"lesser", false}},
+    {"LessOrEqual", {"lesserOrEqual", false}},
+    {"Log", {"log", false}},
+    {"LpPool", {"l2Pool2d", false}},
+    {"MatMul", {"matmul", false}},
+    {"Max", {"max", true}},
+    {"MaxPool", {"maxPool2d", true}},
+    {"Min", {"min", true}},
+    {"Mul", {"mul", true}},
+    {"Neg", {"neg", true}},
+    {"Not", {"logicalNot", false}},
+    {"Pad", {"pad", true}},
+    {"Pow", {"pow", true}},
+    {"PRelu", {"prelu", true}},
+    {"Reciprocal", {"reciprocal", false}},
+    {"ReduceL1", {"reduceL1", false}},
+    {"ReduceL2", {"reduceL2", false}},
+    {"ReduceLogSum", {"reduceLogSum", false}},
+    {"ReduceLogSumExp", {"reduceLogSumExp", false}},
+    {"ReduceMax", {"reduceMax", false}},
+    {"ReduceMean", {"reduceMean", true}},
+    {"ReduceMin", {"reduceMin", false}},
+    {"ReduceProd", {"reduceProduct", false}},
+    {"ReduceSum", {"reduceSum", true}},
+    {"ReduceSumSquare", {"reduceSumSquare", false}},
+    {"Relu", {"relu", true}},
+    {"Reshape", {"reshape", true}},
+    {"Resize", {"resample2d", true}},
+    {"Shape", {"slice", true}},
+    {"Sigmoid", {"sigmoid", true}},
+    {"Softplus", {"softplus", false}},
+    {"Softsign", {"softsign", false}},
+    {"Sin", {"sin", false}},
+    {"Slice", {"slice", true}},
+    {"Softmax", {"softmax", true}},
+    {"Split", {"split", true}},
+    {"Sqrt", {"sqrt", false}},
+    {"Squeeze", {"squeeze", false}},
+    {"Sub", {"sub", true}},
+    {"Tan", {"tan", false}},
+    {"Tanh", {"tanh", true}},
+    {"Transpose", {"transpose", true}},
+    {"Unsqueeze", {"unsqueeze", false}},
+    {"Where", {"elementwiseIf", false}},
 };
 
-inline bool CheckSingleOp(const std::string& op_type, const emscripten::val& wnn_builder_) {
-  return op_map.find(op_type) != op_map.end() && wnn_builder_[op_map.find(op_type)->second].as<bool>();
+inline bool CheckSingleOp(const std::string& op_type, const emscripten::val& wnn_builder_,
+                          const WebnnDeviceType device_type) {
+  // Returns false if the op_type is not listed in the op_map.
+  if (op_map.find(op_type) == op_map.end()) {
+    return false;
+  }
+  // Returns false if the WebNN op has not been implemented in MLGraphBuilder in current browser.
+  if (!wnn_builder_[op_map.find(op_type)->second.opName].as<bool>()) {
+    return false;
+  }
+  // Current WebNN CPU (XNNPack) backend supports limit op list, fallbacks unsupported ops
```

Comment:
"...a limited..."

How about:
> The current WebNN CPU (XNNPack) backend supports a limited op list, and we'd rather fall back early to the ORT CPU EP rather than fail in the WebNN "cpu" deviceType. This is a workaround because ...

Reply:
Done.